### PR TITLE
Support Go modules in SDK

### PR DESCRIPTION
**Pull request by dennwc**
This is the second step to finish the transition to Go modules.

PR bumps major SDK version to `v3`, switches all import paths from `gopkg.in` to Github, fixes docs, etc.

Also, the major version bump allows removing some deprecated symbols in packages. Similarly, take a chance to rename a `uast/yaml` (`uastyml`) package to `uast/uastyaml` (`uastyaml`) to match the package name and import path.

### CODE REVIEW

File: uast/uastyaml/uastyaml.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
+package uastyaml
```
Comment (creachadair) [inline uast/uastyaml/uastyaml.go]: This is just the renamed file, right? (I wish GitHub distinguished moves better)

File: etc/skeleton/driver/normalizer/annotation.go [hunk 2]
```diff
@@ -19,15 +18,6 @@ var Native = Transformers([][]Transformer{
 	},
 }...)
 
-// Code is a special block of transformations that are applied at the end
```
[the remark this answers is not in the document]
Comment (dennwc) [reply etc/skeleton/driver/normalizer/annotation.go]: It's already in the separate commit. I can split them out of the PR of course.

File: uast/uastyaml/uastyaml.go (hunk 1 above)
Comment (dennwc) [reply uast/uastyaml/uastyaml.go]: Correct. I also made temporary aliases in the old package.

File: etc/skeleton/driver/normalizer/annotation.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (creachadair) [reply etc/skeleton/driver/normalizer/annotation.go]: Ah, yeah sorry, I'm still used to squashing commits and we don't do that. Carry on. 😁

[the remark this answers is not in the document]
Comment (dennwc) [reply etc/skeleton/driver/normalizer/annotation.go]: I usually squash everything before opening a PR :)

**Comment by dennwc** [comment]
@bzz Friendly ping

File: .travis.yml [hunk 3]
```diff
@@ -8,19 +8,20 @@ go:
...
+env:
+  - GO111MODULE=on
 
 before_install:
-  - cd $GOPATH/src/gopkg.in/bblfsh/sdk.v2
+  - cd $GOPATH/src/github.com/bblfsh/sdk
   - make validate-commit
-  - wget -O dep https://github.com/golang/dep/releases/download/v0.4.1/dep-linux-amd64 && chmod +x dep
-  - ./dep ensure --vendor-only
+  - go mod download
```
Comment (bzz) [inline .travis.yml]: a bit hard to understand how valuable is to download that before build/test  (e.g could be very valuable if it's warming up a cache for multiple runs but not sure if that is the case)

File: .travis.yml [hunk 4]
```diff
@@ -8,19 +8,20 @@ go:
...
   fast_finish: true
   allow_failures:
     - go: tip
 
+env:
+  - GO111MODULE=on
 
 before_install:
-  - cd $GOPATH/src/gopkg.in/bblfsh/sdk.v2
+  - cd $GOPATH/src/github.com/bblfsh/sdk
```
Comment (bzz) [inline .travis.yml]: not sure why is this still needed?

File: .travis.yml (hunk 3 above)
Comment (dennwc) [reply .travis.yml]: For now, it only separates the dependency download process from the test, nothing more. But if I'm not mistaken, there is a way to cache data across builds. So later we can add Go modules cache to it, for example.

File: .travis.yml (hunk 4 above)
Comment (dennwc) [reply .travis.yml]: In theory, it shouldn't matter anymore. But I would prefer to first make it work like this, and then start cutting up unneeded things. There may be some other scripts that rely on this behavior, for example.

File: etc/skeleton/driver/normalizer/annotation.go [hunk 5]
```diff
@@ -19,15 +18,6 @@ var Native = Transformers([][]Transformer{
...
-// and can access original source code file. It can be used to improve or
-// fix positional information.
-//
-// https://godoc.org/gopkg.in/bblfsh/sdk.v2/uast/transformer/positioner
-var Code = []CodeTransformer{
```
Comment (bzz) [inline etc/skeleton/driver/normalizer/annotation.go]: If this means that on a update of the driver this needs to be removed - let's add it to the SDK release notes, so a driver maintainer is not confused. WDYT?

Comment (dennwc) [reply etc/skeleton/driver/normalizer/annotation.go]: Sure, will mention it. Drivers should already use a new version of this transform stage, so it's only a matter of removing it in the driver as well.

**Pull request merged**